Implement predicates in JDBC DB-Meta class

pjarra opened the pull request:
These are individual commits all referencing #6781.
Support is added for queries to all remaining supports*-predicates in
DuckDBDatabaseMetaData.

pjarra committed:
Don't throw on supportsBatchUpdates, #6781

Batch updates are currently not supported.
Ref: DuckDBPreparedStatement.addBatch().

pjarra committed:
Don't throw on supportsStatementPooling, #6781

Statement pooling is currently not supported by the driver

pjarra committed:
Implement supportsStoredFunctionsUsingCallSyntax, #6781

Citing the jdbc spec:
"If a database supports stored procedures,
 they can be invoked using JDBC escape syntax ..."

DuckDB does not support stored procedures.

pjarra committed:
Implement supportsSchemasInPrivilegeDefs, #6781

Does not apply to DuckDB, hence unsupported

pjarra committed:
Implement supportsResultSetHoldability, #6781

Setting holdability is not supported

Mause commented on tools/jdbc/src/main/java/org/duckdb/DuckDBDatabaseMetaData.java:
```diff
@@ -228,17 +228,17 @@ public boolean supportsConvert() throws SQLException {
 	@Override
 	public boolean supportsConvert(int fromType, int toType) throws SQLException {
-		throw new SQLFeatureNotSupportedException("supportsConvert");
+                throw false;
```
```suggestion
                return false;
```

Mause commented on tools/jdbc/src/main/java/org/duckdb/DuckDBDatabaseMetaData.java:
```diff
@@ -228,17 +228,17 @@ public boolean supportsConvert() throws SQLException {
 	@Override
 	public boolean supportsTableCorrelationNames() throws SQLException {
-		throw new SQLFeatureNotSupportedException("supportsTableCorrelationNames");
+                throw true;
```
```suggestion
                return true;
```

Mause commented on tools/jdbc/src/main/java/org/duckdb/DuckDBDatabaseMetaData.java:
```diff
@@ -228,17 +228,17 @@ public boolean supportsConvert() throws SQLException {
 	@Override
 	public boolean supportsDifferentTableCorrelationNames() throws SQLException {
-		throw new SQLFeatureNotSupportedException("supportsDifferentTableCorrelationNames");
+                throw false;
```
```suggestion
                return false;
```

pjarra committed:
Don't throw on supports*TableCorrelationNames, #6781

- Correlation names (aliases) are supported by DuckDB

- Correlation names are not required to be different from the table name

Reference:
https://duckdb.org/docs/sql/query_syntax/from
https://docs.oracle.com/javase/7/docs/api/java/sql/DatabaseMetaData.html

pjarra committed:
Don't throw on supportsConvert, #6781

This method checks for support of the CONVERT scalar function,
which is not supported by DuckDB. Clients should be able to check
if the database supports this.

pjarra commented:
@Mause sorry, fixed these.

Mytherin commented:
Thanks!
